lambda-runtime/src/main/scala/feral/lambda/runtime/LambdaRuntime.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda
+package runtime
+
+import cats.effect.Temporal
+import cats.syntax.all._
+import cats.effect.syntax.all._
+import cats.effect.kernel.Resource
+import io.circe.Json
+import org.http4s.Method.POST
+import org.http4s.client.Client
+import org.http4s.circe._
+import org.http4s._
+import io.circe._
+
+object LambdaRuntime {
+
+  final val ApiVersion = "2018-06-01"
+
+  def apply[F[_]](client: Client[F])(
+      handlerResource: Resource[F, (Json, Context[F]) => F[Json]])(
+      implicit F: Temporal[F],
+      env: LambdaRuntimeEnv[F]): F[Nothing] =
+    handlerResource.attempt.use[Nothing] { handlerOrError =>
+      env.lambdaRuntimeApi.flatMap[Nothing] { api =>
```

Comment:
Can I ask why `Nothing` is used instead of `Unit` in places such as this? With this code, the compiler complains with 
```
type mismatch;
 found   : Either[Throwable,feral.lambda.runtime.LambdaSettings] => F[Nothing]
 required: Either[Throwable,feral.lambda.runtime.LambdaSettings] => F[B]
```
but runs fine with `Unit`

Reply:
Good question. Technically we should return `Nothing` here because these methods never return—they start a process that runs forever, and will never produce an output. But perhaps type inference is too uncooperative for this to be practical 🤔

Reply:
For example, see `Resource#useForever`:
https://github.com/typelevel/cats-effect/blob/52b5a3ba1623ff40fc48318c9a0bfae558f0e5ed/kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala#L217-L224

Reply:
I think you should be able to use `Nothing` (and should use it, because, as @armanbilge says, it's the right thing to do).
But you probably need to tweak the variance of `F[_]`. My hunch is that it should be declared as `F[+_]`, so
```scala
private[this] def handleInitError[F[+_]](
```
and similarly for others.
Good luck :crossed_fingers:

Reply:
Thanks! I also suspect that `F[+_]` would help things. However type signatures of this sort are sort of an anti-pattern in the Typelevel ecosystem (I'm not aware of any other major projects using them) partly because they prevent the use of datatypes that are not declared covariant at the type-level. To be fair, this might not matter in practice as `F[_]` is almost always going to be `IO` ... but I'd rather not constraint the API here when a few explicit type parameters can do the trick, albeit annoying.

Reply:
> anti-pattern

Is it really, though? AFAIK, maybe contrary to popular opinion, variance is fairly widely used in the TypeLevel world, isn't it? For example IO in Cats Effect is most definitely declared as `IO[+A]`, as you say, and there are surely many other examples.
But fair enough.

But if you @scott-thomson239 don't want to dive into variance, you shouldn't fix it by using `Unit`. The proper way then is to declare additional type variable which can then by inferred to anything that's necessary, e.g.
```scala
  private[this] def handleInitError[F[_], X](runtimeUri: Uri, client: Client[F], ex: Throwable)(
      implicit F: Temporal[F]): F[X] = {
```

Reply:
Okay no problem, I will try and get `Nothing` working instead of `Unit`. I was just using `Unit` temporarily so that I could fix the failing tests so I will revert that.